readthedocs/restapi/utils.py

```diff
@@ -44,18 +48,28 @@ def sync_versions(project, versions, type):  # pylint: disable=redefined-builtin
 
                 log.info(
                     '(Sync Versions) Updated Version: [%s=%s] ',
-                    version['verbose_name'],
-                    version['identifier'],
+                    version_name,
+                    version_id,
                 )
         else:
             # New Version
             created_version = Version.objects.create(
                 project=project,
                 type=type,
-                identifier=version['identifier'],
-                verbose_name=version['verbose_name'],
+                identifier=version_id,
+                verbose_name=version_name,
             )
             added.add(created_version.slug)
+    if not has_user_stable:
+        stable_version = (
+            project.versions
+            .filter(slug=STABLE, type=type)
+            .first()
```

Comment:
Instead of `.first` you can use `.exists` here, since it's exactly what we need.

Reply:
But we need the object on the next line